docs/usage.md

```diff
@@ -37,28 +40,75 @@ $ ./bin/ipc-agent subnet send-value --subnet <subnet-id> --to <to-addr> <value>
+
+Complex behavior can be implemented using these primitives: sending value to a user in another subnet can be implemented a set of `release/fund` and `sendValue` operations. Calling  smart contract from one subnet to another works by providing funds to one account in the destination subnet, and then calling the contract. The agent doesn't currently include abstractions for this complex operations, but it will in the future. That being said, users can still leverage the agent's API to easily compose the basic primitives into complex functionality.
+
+>💡 All cross-net operations need to pay an additional cross-msg fee (apart from the gas cost of the message). This is reason why even if you sent `X FIL` you may see `X - fee FIL` arriving to you account at destination. This fee is used to reward subnet validators for their work committing the checkpoint that carries the message.
+
+### Fund
+Funding a subnet can be performed by using the following command:
 ```bash
-$ ./bin/ipc-agent subnet leave --subnet <subnet-id>
+$ ./bin/ipc-agent cross-msg fund --subnet=<subnet-id> <amount>
 
 # Example execution
-$ ./bin/ipc-agent subnet leave --subnet /root/t01002
+$ ./bin/ipc-agent cross-msg fund --subnet=/root/t01002 100
+
 ```
-Leaving a subnet will release the collateral for the validator and remove all the validation rights from its account. This means that if you have a validator running in that subnet, its validation process will immediately terminate.
+This command includes the cross-net message into the next top-down checkpoint after the current epoch. Once the top-down checkpoint is committed, you should see the funds in your account of the child subnet.
+
+>💡 Top-down checkpoints are not used to anchor the security of the parent into the child (as is the case for bottom-up checkpoints). They just include information of the top-down messages that need to be executed in the child subnet, and are a way for validators in the subnet to reach consensus on the finality on their parent.
+
+### Release
+In order to release funds from a subnet, your account must hold enough funds inside it. Releasing funds to the parent subnet can be permformed with the following comand:
+```bash
+$ ./bin/ipc-agent cross-msg release --subnet=<subnet-id> <amount>
 
+# Example execution
+$ ./bin/ipc-agent cross-msg release --subnet=/root/t01002 100
+```
+This command includes the cross-net message into a bottom-up checkpoint after the current epoch. Once the bottom-up checkpoint is committed, you should see the funds in your account in the parent. 
 
-### Listing checkpoints from a subnet
+
+## Listing checkpoints from a subnet
 
 Subnets are periodically committing checkpoints to their parent every `check-period` (parameter defined when creating the subnet). If you want to inspect the information of a range of checkpoints committed in the parent for a subnet, you can use the `checkpoint list` command provided by the agent as follows: 
```

Comment:
```suggestion
Subnets are periodically committing checkpoints to their parent every `bottomup-check-period` (parameter defined when creating the subnet). If you want to inspect the information of a range of bottom-up checkpoints committed in the parent for a subnet, you can use the `checkpoint list` command provided by the agent as follows: 
```

Reply:
We could consider renaming this `bottomup-list`. Very minor but might be confusing.

Reply:
I already changed in the codebase but was waiting to see if someone complaint, you did so I changed it also in cli and will push the fix in a minute :)